cl/api/tests.py

```diff
@@ -968,3 +971,72 @@ def test_lacks_access(self) -> None:
             r = self.client.get(path)
             self.assertEqual(r.status_code, HTTP_403_FORBIDDEN)
             print("✓")
+
+
+class WebhooksProxySecurityTest(TestCase):
+    """Test Webhook proxy security"""
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.user_profile = UserProfileWithParentsFactory()
+        cls.webhook_https = WebhookFactory(
+            user=cls.user_profile.user,
+            event_type=WebhookEventType.DOCKET_ALERT,
+            url="https://127.0.0.1:3000",
+            enabled=True,
+        )
+
+        cls.webhook_http = WebhookFactory(
+            user=cls.user_profile.user,
+            event_type=WebhookEventType.DOCKET_ALERT,
+            url="http://127.0.0.1:3000",
+            enabled=True,
+        )
+        cls.webhook_0_0_0_0 = WebhookFactory(
+            user=cls.user_profile.user,
+            event_type=WebhookEventType.DOCKET_ALERT,
+            url="http://0.0.0.0:5050",
+            enabled=True,
+        )
+
+    def test_avoid_sending_webhooks_to_internal_ips(self):
+        """Can we avoid sending webhooks to internal IPs?"""
+
+        webhook_event = WebhookEventFactory(
+            webhook=self.webhook_https,
+            content="{'message': 'ok_1'}",
+            event_status=WEBHOOK_EVENT_STATUS.IN_PROGRESS,
+        )
+        send_webhook_event(webhook_event)
+        webhook_to_compare = WebhookEvent.objects.get(pk=webhook_event.pk)
```

Comment:
Can't you just do `webhook_event.refresh_from_db()`?

Reply:
yeah, this work well here, thanks!